### PR TITLE
Backpack: make indices unique

### PR DESCRIPTION
**Pull request by molly-moen**
Update backpack to have unique indexes on user_id and storage_app_id, as we want to mandate uniqueness on both of those columns.

## Links

- spec: [Backpack](https://docs.google.com/document/d/1Y01X2_rrVt8f4y8yHpm3BBpq31Ra9CSkOLYycwQ5or8/edit#heading=h.9zfpephbycva)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/db/migrate/20210722162824_update_backpack_indices.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class UpdateBackpackIndices < ActiveRecord::Migration[5.2]
+  def change
+    remove_index :backpacks, :user_id
```
Comment (bethanyaconnor) [inline dashboard/db/migrate/20210722162824_update_backpack_indices.rb:3]: Up to you if you want to fix this but I _think_ this migration will fail if it needs to be rolled back for any reason. You can split this into an `up` and `down` function to mitigate that. It's pretty unlikely it'll be an issue though

**Review by bethanyaconnor — APPROVED** [review]

File: dashboard/db/migrate/20210722162824_update_backpack_indices.rb (hunk 1 above)
Comment (molly-moen) [reply dashboard/db/migrate/20210722162824_update_backpack_indices.rb:3]: I tested rolling back just now and it worked without any changes!

Comment (bethanyaconnor) [reply dashboard/db/migrate/20210722162824_update_backpack_indices.rb:3]: Yay awesome! Thanks for checking and sorry for the incorrect advice!

Comment (molly-moen) [reply dashboard/db/migrate/20210722162824_update_backpack_indices.rb:3]: no problem, definitely good to double check

**Pull request merged**